era5cli/fetch.py

```diff
@@ -161,6 +167,34 @@ def __init__(
         """bool: Whether to download from the ERA5-Land
         dataset."""
 
+        if self.merge and self.splitmonths:
+            raise ValueError(
+                "\nThe commands '--merge' and '--splitmonths' are not compatible with"
+                "\neach other. Please pick one of the two."
+            )
+
+        if not splitmonths and self.period == "hourly" and (land or ensemble):
+            logging.warning(
+                "\n  The flag --splitmonths was not used, however, in a future version"
+                "\n  this flag will represent the default behavior for hourly --land"
+                "\n  and --ensemble requests. Please update your workflow accordingly."
+            )
+
```

Comment:
If we change the default for a future version, it might make sense to change it for all requests (for consistency), don't you agree?